File: rsa_security.py
import base64
import random
from Crypto.Cipher import AES


def add_to_16(data: str) -> bytes:
    data_bytes = data.encode('utf-8')
    if len(data) % 16:
        add = 16 - (len(data_bytes) % 16)
    else:
        add = 0
    return data_bytes + (b'\0' * add)


def encrypt_by_aes(data: str, iv: str, key: str = 'root%$#@!1234567') -> str:
    key = key.encode('utf-8')
    data = add_to_16(data)
    cryptos = AES.new(key, AES.MODE_CBC, iv.encode('utf-8'))
    cipher_data = cryptos.encrypt(data)
    return base64.standard_b64encode(cipher_data).decode('utf-8')+iv


def decrypt_by_aes(text: str, key: str = 'root%$#@!1234567'):
    key = key.encode('utf-8')
    iv = text[-16:]
    data = text[:-16]
    data = data.encode('utf-8')
    data = base64.b64decode(data)
    cryptos = AES.new(key, AES.MODE_CBC, iv.encode('utf-8'))
    cipher_data = cryptos.decrypt(data)
    return cipher_data.decode('utf-8').strip('\0')


def get_rand_num() -> str:
    digits = [str(i) for i in range(10)]
    lowers = [chr(i) for i in range(65, 91)]
    uppers = [chr(i) for i in range(97, 123)]
    chars = digits+lowers+uppers
    nums = ''
    for i in range(16):
        idx = random.randint(0, 61)
        nums += chars[idx]
    return nums


if __name__ == '__main__':
    data = 'hello'
    iv = get_rand_num()
    text = encrypt_by_aes(data, iv)
    print(text)
    temp = decrypt_by_aes(text)
    assert temp == data

File: rsa-security.js
import CryptoJS from "crypto-js";
function stringtoaesencrypt(data) {
	var temp_key = 'root%$#@!1234567';
	var temp_iv = getRandNum();
	var key = CryptoJS.enc.Latin1.parse(temp_key.substring(0, 16));
	var iv = CryptoJS.enc.Latin1.parse(temp_iv.substring(0, 16));
	var encrypted = CryptoJS.AES.encrypt(data, key, { iv: iv, mode: CryptoJS.mode.CBC, padding: CryptoJS.pad.ZeroPadding });
	return encrypted + temp_iv;
}
function stringtoaesdecrypt(data) {
	var key = 'root%$#@!1234567';
	var iv = data.substring(data.length - 16, data.length);
	var subdata = data.substring(0, data.length - 16);
	var key1 = CryptoJS.enc.Latin1.parse(key);
	var iv1 = CryptoJS.enc.Latin1.parse(iv);
	var decrypted = CryptoJS.AES.decrypt(subdata, key1, {
		iv: iv1,
		mode: CryptoJS.mode.CBC,
		padding: CryptoJS.pad.ZeroPadding
	});
	return decrypted.toString(CryptoJS.enc.Utf8);
}
function getRandNum() {
	var chars = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'];
	var nums = "";
	for (var i = 0; i < 16; i++) {
		var id = parseInt(Math.random() * 61);
		nums += chars[id];
	}
	return nums;
}
console.log(stringtoaesencrypt('hello'))